### PR TITLE
Fix setting WindowState before showing Window.

### LINKED ISSUE #8869: WindowState won't work if window's size is set
**Describe the bug**
If I set the size (`Width` or `Width`) and the `WindowState` of a window before `Show()`, the `WindowState` will return to `Normal` after `Show()`.

**To Reproduce**
Change the code of the SandBox to this:
```
using Avalonia;
using Avalonia.Controls.ApplicationLifetimes;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;

namespace Sandbox
{
    public class App : Application
    {
        public override void Initialize()
        {
            AvaloniaXamlLoader.Load(this);
        }

        public override void OnFrameworkInitializationCompleted()
        {
            if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktopLifetime)
            {
                Window window = new Window();
                window.Width = 1000;
                window.WindowState = WindowState.Maximized;
                window.Show();
            }
        }
    }
}
```
Run the SandBox, the window will show in `Normal` state.

**Expected behavior**
The window should be maximized.

**Desktop (please complete the following information):**
Windows 10 19044
Avalonia 10.18 (I can't compile the latest version)

**Additional context**
If `Width` or `Height` are not set or set the `WindowState` after `Show()`, the `WindowState`  will take effect.

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

As described in #8869, on Win32 if one sets the `Window` (`Min`/`Max`)/`Width`/`Height` together with `WindowState` before showing the window, the `Maximized`/`FullScreen` `WindowState` is ignored.

This PR changes the win32 backend to ignore resize requests when the window is maximized or fullscreen.

## Fixed issues

Fixes #8869

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024896-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0025172-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0025183-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Windows/Avalonia.Win32/WindowImpl.cs [hunk 1]
```diff
@@ -562,6 +567,9 @@ public IRenderer CreateRenderer(IRenderRoot root)
 
         public void Resize(Size value, PlatformResizeReason reason)
         {
+            if (WindowState is WindowState.Maximized or WindowState.FullScreen)
```
Comment (danwalmsley) [inline src/Windows/Avalonia.Win32/WindowImpl.cs:570]: Or minimized? 

!= Normal?

**Review by danwalmsley — CHANGES_REQUESTED** [review]

**Review by danwalmsley — APPROVED** [review]

**Comment by grokys** [comment]
Backported to 0.10.x: ffe74ceaa49b7be878c006c030422bf2d5b0fa73

**Pull request merged**